sql-plugin/src/main/spark311/scala/org/apache/spark/rapids/shims/GpuShuffleExchangeExec.scala

```diff
@@ -47,7 +48,9 @@ case class GpuShuffleExchangeExec(
     gpuOutputPartitioning: GpuPartitioning,
     child: SparkPlan,
     shuffleOrigin: ShuffleOrigin)(
-    cpuOutputPartitioning: Partitioning)
+    cpuOutputPartitioning: Partitioning,
+    val advisoryPartitionSize: Option[Long] = None
```

Comment:
I think advisoryPartitionSize should be in the list of primary arguments to the case class here, as that's what Spark 3.5 is doing.  If there's a reason to leave them in the supplemental arguments then `otherCopyArgs` needs to be updated to handle it, otherwise a copy will either fail or silently drop any value set for it in the new copy.